modules/Microsoft.Network/networkWatchers/.test/common/deploy.test.bicep

```diff
@@ -0,0 +1,144 @@
+    virtualNetworkName: 'dep-<<namePrefix>>-vnet-${serviceShort}'
+    location: location
+  }
+}
+
+// Diagnostics
+// ===========
+module diagnosticDependencies '../../../../.shared/dependencyConstructs/diagnostic.dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep<<namePrefix>>diasa${serviceShort}01'
+    logAnalyticsWorkspaceName: 'dep-<<namePrefix>>-law-${serviceShort}'
+    eventHubNamespaceEventHubName: 'dep-<<namePrefix>>-evh-${serviceShort}'
+    eventHubNamespaceName: 'dep-<<namePrefix>>-evhns-${serviceShort}'
+    location: location
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+#disable-next-line no-hardcoded-location // Disabled as the default RG & location are created in always one location, but each test has to deploy into a different one
+var testLocation = 'westeurope'
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: 'NetworkWatcher_${testLocation}'
+    location: testLocation
+    connectionMonitors: [
+      {
+        name: 'adp-<<namePrefix>>-conmon-${serviceShort}-x-001'
+        endpoints: [
+          {
+            name: '<<namePrefix>>-subnet-x-001(${resourceGroup.name})'
+            resourceId: resourceGroupResources.outputs.virtualMachineResourceId
+            type: 'AzureVM'
+          }
+          {
+            address: 'www.office.com'
+            name: 'Office Portal'
+            type: 'ExternalAddress'
+          }
+        ]
+        testConfigurations: [
+          {
+            httpConfiguration: {
+              method: 'Get'
+              port: 80
+              preferHTTPS: false
+              requestHeaders: []
+              validStatusCodeRanges: [
+                '200'
+              ]
+            }
+            name: 'HTTP Test'
+            protocol: 'Http'
+            successThreshold: {
+              checksFailedPercent: 5
+              roundTripTimeMs: 100
+            }
+            testFrequencySec: 30
+          }
+        ]
+        testGroups: [
+          {
+            destinations: [
+              'Office Portal'
+            ]
+            disable: false
+            name: 'TestHTTPBing'
+            sources: [
+              '<<namePrefix>>-subnet-x-001(${resourceGroup.name})'
+            ]
+            testConfigurations: [
+              'HTTP Test'
+            ]
+          }
+        ]
+        workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+      }
+    ]
+    flowLogs: [
+      {
+        enabled: false
+        storageId: diagnosticDependencies.outputs.storageAccountResourceId
+        targetResourceId: resourceGroupResources.outputs.firstNetworkSecurityGroupResourceId
+      }
+      {
+        formatVersion: 1
+        name: 'adp-<<namePrefix>>-nsg-x-apgw-flowlog'
```

Comment:
Please update with new convention

Reply:
done

Reply:
Is this a dependency or a sub resource? If a dependency, shouldn't it be referenced from the dependencies module output?

Reply:
I just kept the name of the original test :D I guess it's not a dependency then, which makes the name a bit odd. Will double check

Reply:
So there is no such dependency. Renamed it now to something non-dependency looking and updated the readme

Reply:
Those look like child resource names. I'd suggest <<namePrefix>>-${short}-fl-001 and <<namePrefix>>-${short}-cm-001. Not sure why appgw is mentioned

Reply:
True, they are the children - deployed via child modules.

Reply:
Updated the names